vispy/plot/plotwidget.py

```diff
@@ -243,13 +243,19 @@ def mesh(self, vertices=None, faces=None, vertex_colors=None,
         meshdata : MeshData | None
             Meshdata to use. If not None, then vertices, faces, and fname
             must be None.
+        shading : str | 'auto'
+            Shading to use. Default shading is set to 'smooth'. If face_colors
+            is set, shading is set to None.
 
         Returns
         -------
         mesh : instance of Mesh
             The mesh.
         """
         self._configure_3d()
+        shading = 'smooth'
+        if face_colors is not None:
+            shading = None
```

Comment:
think about what happens if a user passes a `shading` param, you are ignoring it here

nest this in an `if shading == 'auto'` and you should be good